File: todo-app/todo-api-java/src/main/java/ca/architech/todo/models/TodoItem.java
package ca.architech.todo.models;

import ca.architech.todo.models.Priority;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "todos")
public class TodoItem {
    private String id;
    private String owner;
    private String description;
    private boolean done;
    private Priority priority;
    private LocalDate dueDate;
    private List<String> tags;

    public void addTag(String tag) {
        tags.add(tag);
    }

}

File: todo-app/todo-api-kotlin/src/main/kotlin/ca/architech/todo/api/TodoItem.kt
package ca.architech.todo.api

import com.fasterxml.jackson.annotation.JsonFormat
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDate

@Document(collection = "todos")
data class TodoItem(var id: String,
                    var owner: String,
                    var done: Boolean = false,
                    var description: String,
                    var priority: Priority,
                    @JsonFormat(pattern = "yyyy-MM-dd")
                    var dueDate: LocalDate,
                    var tags: MutableList<String>) {

    constructor() : this(
            id = "",
            owner = "",
            description = "",
            priority = Priority.NONE,
            dueDate = LocalDate.now(),
            done = false,
            tags = mutableListOf<String>()
    )

    fun addTag(tag: String) {
        tags.add(tag)
    }
}